### DISCUSSION TITLE
REST API deprecated?

### DISCUSSION THREAD

**Discussion by FomTarro** [zepp-health/zeppos-forum#194] [Q&A]
Hello! I build and publish [a software that allows VTubers to use their heart rate as a tracking parameter](https://skeletom-ch.itch.io/vts-heartrate), thereby allowing their model to get flushed or out-of-breath in real time. Today, one user asked me if I could add Zepp Life support. I did some searching and found [an official guide for registering as an OAUTH App](https://github.com/zepp-health/rest-api) and receiving credentials to make REST API calls. However, the application registration link no longer resolves, and the other steps listed in this guide no longer seem to function.

Is there any way that I can apply for OAUTH credentials to do REST calls and fetch heartrate data, giving my users what they want?

**Comment by MyoungXUE** [comment]
Hi, thanks for your interest in us, our rest-api is currently open for business cooperation only.

I have analyzed your requirement and it can be implemented by Zepp OS Mini Program, you can refer to Zepp OS Mini Program architecture design [https://docs.zepp.com/docs/guides/architecture/arc/](https://docs.zepp.com/docs/guides/architecture/arc/) , the Device App calls the heart rate sensor data and then reports the data to the server through the network capability of the Side App.

**Reply by FomTarro** [reply]
Yeah, that's how I had to implement it for Fitbit as well, it's just much less convenient for everyone involved. In particular, when something goes wrong in the chain, users can't get error logs off their devices.

**Comment by estige** [comment]
@MyoungXUE My company has students who use Zepp products how can we establish the partnership between our platform so that our customers' students can use the integration?
